main.nf

```diff
@@ -285,33 +307,75 @@ if (params.subsample) {
       sra_ch.concat(
           csv_pairs_ch, csv_singles_ch, read_pairs_ch,
           read_singles_ch, fastas_ch, read_paths_ch)
-       .ifEmpty{ exit 1, "No reads provided! Check read input files"}
-       .set{ reads_ch }
+       // .ifEmpty{ exit 1, "No reads provided! Check read input files"}
+       .set{ reads_ch_unchecked }
     } else {
       if (params.fastas) {
         // With fasta files - combine everything that can be trimmed
         sra_ch.concat(
             csv_pairs_ch, csv_singles_ch, read_pairs_ch,
             read_singles_ch, read_paths_ch)
-          .set{ ch_read_files_trimming }
+          .set{ ch_read_files_trimming_unchecked }
       } else {
         // No fasta files - combine everything and error out
         sra_ch.concat(
             csv_pairs_ch, csv_singles_ch, read_pairs_ch,
             read_singles_ch, read_paths_ch)
-          .ifEmpty{ exit 1, "No reads provided! Check read input files"}
-          .set{ ch_read_files_trimming }
+          // .ifEmpty{ exit 1, "No reads provided! Check read input files"}
+          .set{ ch_read_files_trimming_unchecked }
       }
     }
   } else {
 //   Do nothing - can't combine the fastq files and bam files (yet)
       sra_ch.concat(
           csv_pairs_ch, csv_singles_ch, read_pairs_ch,
           read_singles_ch, read_paths_ch)
-        .set{ ch_non_bam_reads }
+        .set{ ch_non_bam_reads_unchecked }
     }
 }
 
+protein_input = params.protein_fastas || params.protein_fasta_paths
+if (!protein_input) {
+  println "in not protein input"
+  if (params.subsample) {
+    subsample_reads_ch_unchecked
+      .ifEmpty{ exit 1, "No reads provided! Check read input files"}
```

Comment:
.ifEmpty{ exit 1, "No reads provided! Check read input files in subsample_reads_ch_unchecked"}